ReactSkia/sdk/OnScreenKeyBoard.cpp

```diff
@@ -0,0 +1,618 @@
+    oskHandle.closeWindow();
+    oskHandle.oskState_=OSK_STATE_INACTIVE;
+}
+
+void  OnScreenKeyboard::updatePlaceHolderString(std::string PHdisplayString) {
+    RNS_LOG_TODO("updatePlaceHolderString:Implementation yet to be done");
+    return;
+}
+
+void OnScreenKeyboard::launchOSKWindow(OSKConfig oskConfig) {
+
+    memcpy(&oskConfig_,&oskConfig,sizeof(oskConfig));
+
+    SkSize mainscreenSize_=RnsShell::PlatformDisplay::sharedDisplay().screenSize();
+    if(screenSize_ != mainscreenSize_) {
+        generateOSKLayout_=true;
+        screenSize_=mainscreenSize_;
+    } else {
+        generateOSKLayout_=false;
+    }
+    //Set up OSk configuration
+    if(oskConfig_.theme == OSK_LIGHT_THEME) {
+        bgColor_ = OSK_LIGHT_THEME_BG_COLOR ;
+        fontColor_ = OSK_LIGHT_THEME_FONT_COLOR;
+    } else {
+        bgColor_ = OSK_DARK_THEME_BG_COLOR ;
+        fontColor_ =OSK_DARK_THEME_FONT_COLOR;
+    }
+    if(oskConfig_.type == OSK_ALPHA_NUMERIC_KB)
+        oskLayout_.kbLayoutType = ALPHA_LOWERCASE_LAYOUT;
+
+    unsigned int XscaleFactor = screenSize_.width()/baseScreenSize.width();
+    unsigned int YscaleFactor =screenSize_.height()/baseScreenSize.height();
+    oskLayout_.textFontSize= OSK_FONT_SIZE *XscaleFactor;
+    oskLayout_.textHLFontSize= OSK_HIGHLIGHT_FONT_SIZE *XscaleFactor;
+    oskLayout_.horizontalStartOffset= ((screenSize_.width()-(screenSize_.width()*OSK_PLACEHOLDER_LENGTH))/2)*XscaleFactor;
+
+    /*Craeting OSK Window*/
+    std::function<void()> createWindowCB = std::bind(&OnScreenKeyboard::windowReadyToDrawCB,this);
+    createWindow(screenSize_,createWindowCB);
+
+}
+
+bool OnScreenKeyboard::drawOSK() {
+
+if((oskState_!= OSK_STATE_ACTIVE)) return false;
+
+/*1. Set up OSK Window background*/
+    oskCanvas->clear(bgColor_);
+
+/*2. Draw PlaceHolder*/
+    SkPaint paint;
+    if(oskConfig_.placeHolderName) {
+        SkFont font;
+        paint.setColor(fontColor_);
+        font.setSize(OSK_FONT_SIZE);
+        oskCanvas->drawSimpleText(oskConfig_.placeHolderName,strlen(oskConfig_.placeHolderName), SkTextEncoding::kUTF8,
+                                   oskLayout_.horizontalStartOffset,
+                                   screenSize_.height()*OSK_PH_NAME_VERTICAL_OFFSET,
+                                   font, paint);
+    }
+    paint.setColor((oskConfig_.theme == OSK_LIGHT_THEME) ? OSK_LIGHT_THEME_PH_COLOR: OSK_DARK_THEME_PH_COLOR);
+    SkRect rect=SkRect::MakeXYWH( oskLayout_.horizontalStartOffset,
+                                  screenSize_.height()*OSK_PLACEHOLDER_VERTICAL_OFFSET,
+                                  screenSize_.width()*OSK_PLACEHOLDER_LENGTH,
+                                  OSK_PLACEHOLDER_HEIGHT
+                                );
+    oskCanvas->drawRect(rect,paint);
+
+/*3. Draw PLaceHolder Display String */
+    RNS_LOG_TODO("Update PlaceHolder String on start Up");
+
+/*4. Draw  KeyBoard Layout*/
+    drawKBLayout(oskConfig_.type);
```

Comment:
return value not checked. even if inactive , further calls are still been done here right